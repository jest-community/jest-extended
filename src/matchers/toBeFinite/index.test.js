import matcher from './';

expect.extend(matcher);

describe('.toBeFinite', () => {
  it('passes when given a positive finite number', () => {
    expect(1).toBeFinite();
  });

  it('passes when given a negative finite number', () => {
    expect(-1).toBeFinite();
  });

  it('passes when given the largest finite number', () => {
    expect(Number.MAX_SAFE_INTEGER).toBeFinite();
  });

  it('passes when given the largetst negative finite number', () => {
    expect(Number.MIN_SAFE_INTEGER).toBeFinite();
  });

  it('passes when given 0', () => {
    expect(0).toBeFinite();
  });

  it('fails when not given NaN', () => {
    expect(() => expect(NaN).toBeFinite()).toThrowErrorMatchingSnapshot();
  });

  it('fails when not given Infinity', () => {
    expect(() => expect(Infinity).toBeFinite()).toThrowErrorMatchingSnapshot();
  });
});

describe('.not.toBeFinite', () => {
  it('fails when given a finite number', () => {
    expect(() => expect(1).not.toBeFinite()).toThrowErrorMatchingSnapshot();
  });

  it('passes when given NaN', () => {
    expect(NaN).not.toBeFinite();
  });

  it('passes when given Infinity', () => {
    expect(Infinity).not.toBeFinite();
  });
});
