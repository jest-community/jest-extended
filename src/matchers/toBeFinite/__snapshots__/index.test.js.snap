// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`.not.toBeFinite fails when given a finite number 1`] = `
"<dim>expect(</><red>received</><dim>).not.toBeFinite(</><dim>)</>

Expected value to not be finite received:
  <red>1</>"
`;

exports[`.toBeFinite fails when not given Infinity 1`] = `
"<dim>expect(</><red>received</><dim>).toBeFinite(</><dim>)</>

Expected value to be finite received:
  <red>Infinity</>"
`;

exports[`.toBeFinite fails when not given NaN 1`] = `
"<dim>expect(</><red>received</><dim>).toBeFinite(</><dim>)</>

Expected value to be finite received:
  <red>NaN</>"
`;
